clients/privacy-center/cypress/e2e/should-show-experience.cy.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  CONSENT_COOKIE_NAME,
+  NoticeConsent,
+  PrivacyExperience,
+} from "fides-js";
+
+import { mockCookie } from "../support/mocks";
+import { OVERRIDE, stubConfig, stubTCFExperience } from "../support/stubs";
+
+describe("Fides.shouldShowExperience()", () => {
+  interface TestCaseOptions {
+    fixture:
+      | "experience_modal.json"
+      | "experience_banner_modal.json"
+      | "experience_tcf_minimal.json"
+      | "experience_none.json";
+    savedConsent: boolean;
+    expiredTcfVersionHash?: boolean;
+    shouldShowExperience: boolean;
+  }
+
+  const testCases: TestCaseOptions[] = [
+    {
+      fixture: "experience_modal.json",
+      savedConsent: false,
+      shouldShowExperience: false,
+    },
+    {
+      fixture: "experience_modal.json",
+      savedConsent: true,
+      shouldShowExperience: false,
+    },
+    {
+      fixture: "experience_banner_modal.json",
+      savedConsent: false,
+      shouldShowExperience: true,
+    },
+    {
+      fixture: "experience_banner_modal.json",
+      savedConsent: true,
+      shouldShowExperience: false,
+    },
+    {
+      fixture: "experience_tcf_minimal.json",
+      savedConsent: false,
+      shouldShowExperience: true,
+    },
+    {
+      fixture: "experience_tcf_minimal.json",
+      savedConsent: true,
+      shouldShowExperience: false,
+    },
+    {
+      fixture: "experience_tcf_minimal.json",
+      savedConsent: true,
+      expiredTcfVersionHash: true,
+      shouldShowExperience: true,
+    },
+    {
+      fixture: "experience_none.json",
+      savedConsent: false,
+      shouldShowExperience: false,
+    },
+    {
+      fixture: "experience_none.json",
+      savedConsent: true,
+      shouldShowExperience: false,
+    },
+  ];
+
+  testCases.forEach(
+    ({
+      fixture,
+      savedConsent,
+      expiredTcfVersionHash,
+      shouldShowExperience,
+    }) => {
+      describe(`when rendering ${fixture} and saved consent ${savedConsent ? "exists" : "does not exist"} ${expiredTcfVersionHash ? "(with expired tcf_version_hash)" : ""}`, () => {
```

Comment:
Thanks to the parameterized tests above, this'll exercise 9 different combinations of experiences + saved consent.